specs/_features/maxeb_increase/capella.py

```diff
@@ -1547,9 +1557,13 @@ def process_registry_updates(state: BeaconState) -> None:
         if is_eligible_for_activation(state, validator)
         # Order by the sequence of activation_eligibility_epoch setting and then index
     ], key=lambda index: (state.validators[index].activation_eligibility_epoch, index))
-    # Dequeued validators for activation up to churn limit
-    for index in activation_queue[:get_validator_churn_limit(state)]:
+    # Dequeue validators for activation up to churn limit [MODIFIED TO BE WEIGHT-SENSITIVE]
+    max_churn_left = get_validator_churn_limit(state) # This is now a Gwei amount
+    for index in activation_queue:
         validator = state.validators[index]
+        max_churn_left -= validator.effective_balance
+        if max_churn_left < 0:
```

Comment:
avoid using negative numbers—rewrite as `if max_churn_left < validator.effective_balance`

Reply:
reorder to 

```
for index in activation_queue:
        validator = state.validators[index]
        if max_churn_left < validator.effective_balance:
            break
        max_churn_left -= validator.effective_balance
        validator.activation_epoch = compute_activation_exit_epoch(get_current_epoch(state))
```